18F/identity-idp
# LG-13014: Implement password compromised

## 🎫 Ticket

Link to the relevant ticket:
[LG-13014](https://cm-jira.usa.gov/browse/LG-13014)

## 🛠 Summary of changes
This will Allow users to be notified that their password has been compromised and prompted to change their password. 

## 📜 Testing Plan

Provide a checklist of steps to confirm the changes.
Steps to test

1.  Set `compromised_password_randomizer_value` and `compromised_password_randomizer_threshold` to 1 and `check_user_password_compromised_enabled` to true in the environment you are testing on. 
2. Create An account with Bad password (Will need to go into rails console and update it to be a bad password after creating account) 
3. Sign out of account and sign in with Bad password
4. Be expected to be taken to a page lettign user know to change password
5.  If you sign out and attempt to do it again you will be directed to `after_sign_in_path` for user instead of password compromised page. 


## 👀 Screenshots
![Screenshot 2024-06-25 at 9 19 30 AM](https://github.com/18F/identity-idp/assets/23225522/6d866d19-248b-44ad-8ccc-7e8e89546758)
